### PR TITLE
Import Snapweb styles into toolkit.

### PR DESCRIPTION
**Pull request by justincan**
- Add Vanilla CSS toolkit
- Add sass processing step
- Import snapweb styles from snapweb project.
- Exporting vanilla based snapweb styles globally

### CODE REVIEW

File: src/styles/styles.scss [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+// Site specific utilities
+@include u-full-width;
+
+// WIP - not quite decided where best to put these styles yet
+// Raised as Vanilla bug: https://github.com/ubuntudesign/vanilla-framework/issues/701
+.row {
+  padding-top: 1rem;
+  padding-bottom: 1rem;
+}
```
Comment (anthonydillon) [inline src/styles/styles.scss:39]: `p-strip` is in fact the pattern that adds vertical padding. `row` is purely a structural element.

The grid's docs does a good job of highlight how to use it correctly: http://shelvesgrid.org/docs/css

File: package.json [hunk 2]
```diff
@@ -33,8 +33,11 @@
     "css-loader": "^0.26.1",
     "extract-text-webpack-plugin": "2.0.0-rc.0",
     "file-loader": "^0.9.0",
+    "node-sass": "^4.5.0",
+    "sass-loader": "^6.0.3",
     "source-map-loader": "^0.1.6",
     "url-loader": "^0.5.7",
+    "vanilla-framework": "1.0.1",
```
Comment (anthonydillon) [inline package.json:40]: Vanillas current version is `1.1.10`

File: src/styles/_patterns_card-deck.scss [hunk 3]
```diff
@@ -0,0 +1,93 @@
+@mixin p-card-deck {
```
Comment (anthonydillon) [inline src/styles/_patterns_card-deck.scss:1]: It's best practise to name space your site specific mixins with a site identifier. For example `sw-p-card-deck`. This helps to stop naming collisions.

File: src/styles/styles.scss [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+// This has been proposed upstream to Vanilla Framework
+// https://github.com/ubuntudesign/vanilla-framework/issues/710
+hr {
+  border: 0;
+  height: 0;
+  border-top: 1px solid $color-mid-light;
+  margin: 2rem 0;
+}
```
Comment (anthonydillon) [inline src/styles/styles.scss:48]: This issue has been resolved with the latest version of Vanilla.

**Review by anthonydillon — COMMENTED** [review]
Good work! Great to see Vanilla being used outside of our team. I had a few suggests inline.

**Comment by justincan** [comment]
Hey @anthonydillon, great feedback. This branch represents a direct import from the snapweb project (hence version, etc). For traceability, I'm going to commit as is. But will make changes in a future PR. Thanks for looking, and please check in on future commits!

**Comment by AlexandreAbreu** [comment]
I am not sure the yarn.lock file is necessary ...

**Comment by justincan** [comment]
The lock file is necessary, see https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

**Comment by AlexandreAbreu** [comment]
LGTM then

**Comment by dbarth** [comment]
+1 for me as well, let's go then.

**Pull request merged**